source/common/router/config_impl.cc

```diff
@@ -385,6 +391,7 @@ void RouteEntryImplBase::finalizeRequestHeaders(Http::HeaderMap& headers,
 
 void RouteEntryImplBase::finalizeResponseHeaders(
     Http::HeaderMap& headers, const RequestInfo::RequestInfo& request_info) const {
+  route_action_response_headers_parser_->evaluateHeaders(headers, request_info);
```

Comment:
Probably adding a comment the same fashion with `finalizeRequestHeaders` here will add clarity.